core/controllers/collection_editor.py

```diff
@@ -102,20 +104,40 @@ def get(self, collection_id):
         collection = collection_services.get_collection_by_id(
             collection_id, strict=False)
 
+        can_edit = (
+            bool(self.user_id) and
+            self.username not in config_domain.BANNED_USERNAMES.value and
+            rights_manager.Actor(self.user_id).can_edit(
+                feconf.ACTIVITY_TYPE_COLLECTION, collection_id))
+
+
+        user_permissions = rights_manager.get_collection_rights(
```

Comment:
I don't think the dict conversion is desirable here. Just get the domain object and refer to collection_rights.owner_names etc.? If you need methods for ids --> names, introduce collection_rights.get_owner_names() etc. on the domain object itself.